### PR TITLE
chore: upgrade @celo sdk

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

This PR upgrades the Celo sdk's. This means that we can continue to use ODIS, and re-enable the secure send tests.

This PR does not include updating the payments sdk, that is still in progress.

### Test plan

- Wallet CI passes

### Related issues

- Fixes RET-637

### CODE REVIEW

File: src/identity/privateHashing.ts [hunk 1]
```diff
@@ -130,17 +127,27 @@ function* getPhoneHashPrivate(e164Number: string, selfPhoneHash?: string) {
...
       blindingFactor,
-      selfPhoneHash,
       DeviceInfo.getVersion(),
-      blsBlindingClient
+      blsBlindingClient,
+      undefined,
+      undefined,
+      CombinerEndpointPNP.LEGACY_PNP_SIGN
     )
+    const phoneNumberHashDetails: PhoneNumberHashDetails = {
```
Comment (kathaypacific) [inline src/identity/privateHashing.ts:144]: making this `getObfuscatedIdentifier` backwards compatible with `getPhoneNumberIdentifier` with the least amount of changes...

File: __mocks__/@celo/identity/index.ts [hunk 2]
```diff
@@ -21,15 +21,11 @@ enum AuthenticationMethod {
...
-      odisUrl: 'alfajoresstaging',
-      odisPubKey: 'alfajoresstaging',
-    },
-    ODIS_ALFAJORES_CONTEXT: {
+    ODIS_ALFAJORES_CONTEXT_PNP: {
       odisUrl: 'alfajores',
       odisPubKey: 'alfajores',
     },
-    ODIS_MAINNET_CONTEXT: {
+    ODIS_MAINNET_CONTEXT_PNP: {
```
Comment (kathaypacific) [inline __mocks__/@celo/identity/index.ts:28]: i did verify that this is the context that returns the values that we were using before

File: src/merchantPayment/hooks.ts [hunk 3]
```diff
@@ -48,6 +48,7 @@ export function useMerchantPayments(apiBase: string, referenceId: string) {
       dispatch(showError(t('merchantWalletUnlockError')))
     }
 
+    // @ts-ignore
```
Comment (kathaypacific) [inline src/merchantPayment/hooks.ts:51]: will remove this when https://github.com/celo-org/payments/pull/35 is merged and we upgrade the payments sdk

**Review by jeanregisser — APPROVED** [review]
Amazing! 🚀 

Looking forward to have the payment SDK updated too, to avoid bundling multiple versions of the Celo SDK.

**Comment by kathaypacific** [comment]
> Amazing! 🚀
> 
> Looking forward to have the payment SDK updated too, to avoid bundling multiple versions of the Celo SDK.

yes just waiting on a new version of the payments sdk to be published :)

**Comment by kathaypacific** [comment]
i want to merge this PR as it is now, and there will be 2 follow ups:
1) i am waiting on a new version of the payments sdk to be published by our cLabs friends, when that is available i will update. that will remove the duplicate versions of the celo packages in our bundle
2) the secure send tests are broken due to a new change to the send flow, waiting to get clarity on the expected behaviour. the test will need to be updated before we can re-enable them again https://valora-app.slack.com/archives/C02KBT0DAHJ/p1677839014227009 so i'm not going to include that here

**Pull request merged**